cmd/zfs/zfs_main.c

```diff
@@ -3466,9 +3468,40 @@ print_header(list_cbdata_t *cb)
 			(void) printf("%-*s", (int)pl->pl_width, header);
 	}
 
+	color_end();
+
 	(void) printf("\n");
 }
 
+/*
+ * Decides on the color that the avail value should be printed in.
+ * > 80% used = yellow
+ * > 90% used = red
+ */
+static const char *
+color_available(zfs_handle_t *zhp)
+{
+	char used_prop[ZFS_MAXPROPLEN];
+	char available_prop[ZFS_MAXPROPLEN];
+
+	zfs_prop_get(zhp, ZFS_PROP_USED, used_prop,
+	    sizeof (used_prop), NULL, NULL, 0, 1);
+
+	zfs_prop_get(zhp, ZFS_PROP_AVAILABLE, available_prop,
+	    sizeof (available_prop), NULL, NULL, 0, 1);
+
+	int used = atoi(used_prop);
+	int available = atoi(available_prop);
+	int percentage = (int)((double)available / (available + used) * 100);
+
+	if (percentage > 20)
+		return ("");
```

Comment:
For the non-colored case, the empty string ("") will get passed to `color_start()`, which is technically fine, but does cause color_start() to do an unnecessary flush().  You could pass back `NULL` instead and just modify color_start() to do nothing if it's passed NULL.